animations/lib/src/misc/curved_animation.dart

```diff
@@ -0,0 +1,146 @@
+  AnimationController animController;
+  Animation<double> animation;
+  static const _duration = Duration(seconds: 4);
+  TextStyle style = TextStyle(fontSize: 18.0);
+  List<MyCurve> forwardCurves = [
+    MyCurve(curve: Curves.bounceIn, curveName: 'Bounce In'),
+    MyCurve(curve: Curves.easeInCubic, curveName: 'Ease In Cubic'),
+    MyCurve(curve: Curves.easeInExpo, curveName: 'Ease In Expo'),
+    MyCurve(curve: Curves.elasticIn, curveName: 'Elastic In'),
+    MyCurve(curve: Curves.easeInQuart, curveName: 'Ease In Quart'),
+    MyCurve(curve: Curves.easeInCirc, curveName: 'Ease In Circ'),
+  ];
+  List<MyCurve> reverseCurves = [
+    MyCurve(curve: Curves.bounceOut, curveName: 'Bounce Out'),
+    MyCurve(curve: Curves.easeOutCubic, curveName: 'Ease Out Cubic'),
+    MyCurve(curve: Curves.easeOutExpo, curveName: 'Ease Out Expo'),
+    MyCurve(curve: Curves.elasticOut, curveName: 'Elastic Out'),
+    MyCurve(curve: Curves.easeOutQuart, curveName: 'Ease Out Quart'),
+    MyCurve(curve: Curves.easeOutCirc, curveName: 'Ease Out Circ')
+  ];
+  MyCurve selectedForwardCurve, selectedReverseCurve;
+  CurvedAnimation curvedAnimation;
+  @override
+  void initState() {
+    super.initState();
+    animController = AnimationController(
+      duration: _duration,
+      vsync: this,
+    );
+    selectedForwardCurve = forwardCurves[0];
+    selectedReverseCurve = reverseCurves[0];
+    curvedAnimation = CurvedAnimation(
+      parent: animController,
+      curve: selectedForwardCurve.curve,
+      reverseCurve: selectedReverseCurve.curve,
+    );
+    animation = Tween<double>(
+      begin: 0,
+      end: 2 * math.pi,
+    ).animate(curvedAnimation)
+      ..addListener(() {
+        setState(() {});
+      })
+      ..addStatusListener((status) {
+        if (status == AnimationStatus.completed) {
+          animController.reverse();
+        }
+      });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: Text('Curved Animation')),
+      body: Column(
+        children: [
+          SizedBox(height: 20.0),
+          Text(
+            'Select Curve for forward rotation',
+            style: style,
+          ),
+          DropdownButton<MyCurve>(
+            items: forwardCurves.map((dropDownItem) {
+              return DropdownMenuItem<MyCurve>(
+                  value: dropDownItem, child: Text(dropDownItem.curveName));
+            }).toList(),
+            onChanged: (newCurve) {
+              setState(() {
+                selectedForwardCurve = newCurve;
+                curvedAnimation.curve = selectedForwardCurve.curve;
+              });
+            },
+            value: selectedForwardCurve,
+          ),
+          SizedBox(height: 15.0),
+          Text(
+            'Select Curve for reverse rotation',
+            style: style,
+          ),
+          DropdownButton<MyCurve>(
+            items: reverseCurves.map((dropDownItem) {
```

Comment:
nit: The variable name `dropDownItem` implies that the value it contains is already a drop down item. Consider using `curve` instead.